access-token-management/test/AccessTokenManagement.Tests/LogExpirationTests.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Duende Software. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using Duende.AccessTokenManagement.Tests;
+using Microsoft.Extensions.Logging;
+
+namespace Duende.AccessTokenManagement;
+public class LogExpirationTests
+{
+    /// <summary>
+    /// This test proves that the source generator for log messages do not invoke the function
+    /// passed as an argument. 
+    /// </summary>
+    [Fact]
+    public void Logging_using_a_function_will_not_invoke_function()
```

Comment:
I tried having log messages using a 'delegate' for lazy message evaluation. Apparently that doesn't work.